### DISCUSSION TITLE
plot_s_db() Add fixed offset

### DISCUSSION THREAD

**Discussion by wktanghk** [scikit-rf/scikit-rf#756] [Q&A] [answered]
Hi, I'm not sure if this is the right place to ask about a possible feature for plot_s_db(). If not, please kindly advise the right place to ask.

I'm hoping to add a fixed offset to my plot_s_db() graph. For example, I have S21 = 5dB across freqeuency (i.e. flat line vs freq), and I hope to add 2dB to it, such that I will get a graph which is a flat line @ 7dB across frequency.

I can do that in ADS by the equation "S21 + 5". I'm wondering if I can do it with plot_s_db() as well. If not, would there be any workaround? Thanks!

**Comment by jhillairet** [comment, marked as answer]
Dear @wktanghk 

This is the perfect place for asking.

Assuming your Network is `ntwk`, and if the purpose is only to plot the S21 + 2dB, the following trick should work:

`(ntwk.s21 * rf.db_2_mag(2)).plot_s_db()  # add 2 dB to S21 and plot the result `

**Reply by wktanghk** [reply]
Hi Julien,

Thank you so much for your timely help, and a very interesting solution! This would almost work for me. For my case, because I need to plot many of these curves, and the input & output ports would need to be determined during run time, so I made them as variables like below. And I would like to add 2dB to all of them (e.g. S21+2dB; S75+2dB; S(13,11)+2dB, etc.). Would it be possible to modify the syntax to support this?

`nw.plot_s_db(m=V, n=A)`

Thanks,
William

**Reply by jhillairet** [reply]
Before adding a new feature (and generally new bugs as well...), let's see if we can find a workaround to your specific problem ;-)

Adding N dB to all the S-parameter of your Network `ntwk` of them could be :

``` python
# Create a N-port Network offset (2 dB for all S-param at all freq)
offset = ntwk.copy()
offset.s[:] = rf.db_2_mag(2)
# Add the offset to all the S-param of your Network
ntwk_offsetted = ntwk * offset
 # plot the offsetted S(V,A) in dB
ntwk.plot_s_db(m=V, n=A, label='original')
ntwk_offsetted.plot_s_db(m=V, n=A, label='offsetted')
```

**Reply by wktanghk** [reply]
It works, thanks! And I'm completely with you that we should avoid adding unnecessary new features (and bugs!)

Thank you so much for your help, and I learned so much along the way. Since this is my first time posting a Discussion, Please let me know if I should mark this as "closed". Thanks!

**Reply by jhillairet** [reply]
Great. If the solution fits to you, yes, you can mark it as closed.

**Reply by wktanghk** [reply]
Thanks @jhillairet . This issue is now closed. I will look to see if there is a button to properly close this on github.
